Bugfix: #548 - Adding delay to throw_item and UnarmedAttack

jack-fractal opened the pull request:
Adding a delay to throw_item and UnarmedAttack click events.

Throw_Item's delay was set to 12, slightly slower than UnarmedAttack. 

Appears to have fixed all of the issues raised in the bug report with the exception of throwing a mob. I couldn't figure out how to do that quickly in the first place.

SoundScopes commented:
At the moment I am considering revering the click delay proc that you added.
Because it was half done, there are loads of places in the code where it needs fixing or changing.

jack-fractal commented:
The click delay proc is literally just a shortcut to a very common pattern which was 'next_move = world.time + x'. I added it because it seemed like a reasonable thing to do, and by placing the various delay times in the setup.dm file, it would let people tweak the various delays to modify the game. 

I didn't update every instance of that in the game because that was well beyond the scope of the change that I wanted to make, but I had a few minutes tonight so I went and did that. There wasn't actually as much as you might imagine, about 35 places.  I also simplified "AllowedToMove" tests by making them a function as well.

SoundScopes commented:
it may have been beyond the scope of what you wanted to change, but it wasn't noted or mentioned and it wasn't required for your feature. So if someone had decided to change the values, it wouldn't have worked in those places. Which in the end makes more work for whoever.

However you did it so, awaiting
